charts/opentelemetry-demo/examples/default/rendered/component.yaml

```diff
@@ -382,32 +382,32 @@ spec:
             name: service
           env:
           
-          - name: OTEL_SERVICE_NAME
-            valueFrom:
-              fieldRef:
-                apiVersion: v1
-                fieldPath: metadata.labels['app.kubernetes.io/component']
-          - name: OTEL_K8S_NAMESPACE
+          - name: "OTEL_K8S_NAMESPACE"
             valueFrom:
               fieldRef:
                 apiVersion: v1
                 fieldPath: metadata.namespace
-          - name: OTEL_K8S_NODE_NAME
+          - name: "OTEL_K8S_NODE_NAME"
             valueFrom:
               fieldRef:
                 apiVersion: v1
                 fieldPath: spec.nodeName
-          - name: OTEL_K8S_POD_NAME
+          - name: "OTEL_K8S_POD_NAME"
             valueFrom:
               fieldRef:
                 apiVersion: v1
                 fieldPath: metadata.name
-          - name: OTEL_RESOURCE_ATTRIBUTES
+          - name: "OTEL_RESOURCE_ATTRIBUTES"
             value: service.name=$(OTEL_SERVICE_NAME),k8s.namespace.name=$(OTEL_K8S_NAMESPACE),k8s.node.name=$(OTEL_K8S_NODE_NAME),k8s.pod.name=$(OTEL_K8S_POD_NAME)
-          - name: OTEL_EXPORTER_OTLP_ENDPOINT
-            value: http://example-otelcol:4317
-          - name: AD_SERVICE_PORT
+          - name: "OTEL_SERVICE_NAME"
```

Comment:
This is the downside to switching the env vars from a list to a map: `range` doesn't preserve order 😞.  But we need order preserved because we need to ensure `OTEL_RESOURCE_ATTRIBUTES` is defined after the env vars it depends on. In order to be able to switch from a list to a map we need to be able to keep order I think.

Reply:
I see, do you know if the order is kept stable inside the map and only the range is spoiling it? So are you confident on that there can be a solution, then I will try out to find something.

Otherwise, we could introduce a deterministic order by sorting and treat the service name special which of course is ugly

Reply:
I'm 99% sure that it a problem with Maps and not `range`.  I think there _might_ be a solution that uses `| sortAlpha`, but that will only sort alphabetically.  We could use it to our advantage, but it feels hacky.

Reply:
As far as I can see Helm dictionaries have no order, so sorting the list of keys will make the order stable and deterministic. Still I don't see how the OTEL_SERVICE_NAME can be moved to the last element without an explicit hack taking care of that.

So either we accept that there can be no dependency between env variables or we accept that you cannot override specific elements (and find a different way to set especially the collector URLs more simple)

Another idea: We already have 2 layers of env variables with the defaultEnvs, we could also say that dependencies can exist between the two layers only and we move the OTEL_SERVICE_NAME down to each component section.

Reply:
We could move `OTEL_RESOURCE_ATTRIBUTES` down, but it breaks the notion of being a default attribute.  If you want to update `OTEL_RESOURCE_ATTRIBUTES` for every component, you'd have to update each component's version vs only changing the instance in `default.env`.

Circling back the the larger issue prompting the PR, when you had to redefine variables, what did your values.yaml end up looking like?  It's annoying to have to redefine lists, but their consistency in these types of situations is useful.

Reply:
I see your point and understand that this is much better from a consistency perspective.

However, I want to provide instructions on how to install the demo app using a different collector infrastructure. For that I want to provide an examplaric values.yaml defining the required overrides. That values.yaml contains close to 200 lines at the moment where I mainly want to reconfigure the collector URL and disable the bundled collector. That 200 lines I would need to check for every update if they are still complete or adjusted.

Going back to the initial discussion https://github.com/open-telemetry/opentelemetry-helm-charts/pull/423#discussion_r995157372, having the collector domain configurable and re-using that variable in the env definitions could be a better solution. However, it will not solve the override problem in a generic way...

Reply:
@a-thaler would you be willing to share your values.yaml in the #438?  I'd love to see the issue first hand so I can experience the pain.

Reply:
I shared my scenario with the values.yaml at https://github.com/open-telemetry/opentelemetry-helm-charts/issues/438#issuecomment-1290237295